actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -319,7 +319,7 @@ def GET
     def POST
       @env["action_dispatch.request.request_parameters"] ||= Utils.deep_munge(normalize_encode_params(super || {}))
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
-      raise ActionController::BadRequest.new(:request, e)
+      raise ActionController::BadRequest, "Invalid request parameters: #{e.message}", e.backtrace
```

Comment:
Another thought on cause handling. Setting an exception's backtrace to its cause's backtrace may no longer be necessary, if exception handlers know to check causes. It's clearer to see the raised exception's backtrace, unobscured, and separately report the backtrace of its cause. That'd require updating our exception handlers and middleware to be cause-aware.